api/src/main/java/xyz/gianlu/librespot/api/handlers/InstanceHandler.java

```diff
@@ -105,20 +130,31 @@ protected void handleRequest(@NotNull HttpServerExchange exchange, @NotNull Sess
                 return;
             }
 
-            String action = getAction(exchange);
-            if (action == null) return;
-
-            switch (action) {
-                case "terminate":
-                    exchange.endExchange();
-                    new Thread(server::stop).start();
-                    break;
-                case "close":
-                    player.close();
-                    session.close();
+            String requestMethod = exchange.getRequestMethod().toString();
+            switch(requestMethod) {
+                case "GET":
+                    String info = getInstanceInfo(session);
+                    exchange.getResponseHeaders().put(Headers.CONTENT_TYPE, "application/json");
+                    exchange.getResponseSender().send(info);
+                    return;
+                case "POST":
+                    String action = getAction(exchange);
+                    if (action == null) return;
+
+                    switch (action) {
+                        case "terminate":
+                            exchange.endExchange();
+                            new Thread(server::stop).start();
+                            break;
+                        case "close":
+                            session.close();
+                            break;
+                        default:
+                            Utils.invalidParameter(exchange, "action");
+                            break;
+                    }
                     break;
                 default:
-                    Utils.invalidParameter(exchange, "action");
```

Comment:
Same 405 here

Reply:
Apologies, would you like 405 inside the requestMethod switch? or the action switch?

Reply:
You're right, good as is them.